lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/TracingCacheManager.java

```diff
@@ -0,0 +1,162 @@
+    {
+        this.tracer = requireNonNull(tracer, "tracer is null");
+        this.cacheKey = requireNonNull(cacheKey, "cacheKey is null");
+        this.delegate = requireNonNull(delegate, "delegate is null");
+    }
+
+    @Override
+    public boolean put(PageId pageId, ByteBuffer byteBuffer, CacheContext cacheContext)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.put")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .setAttribute(CACHE_FILE_WRITE_POSITION, (long) byteBuffer.position())
+                .setAttribute(CACHE_FILE_WRITE_SIZE, (long) byteBuffer.remaining())
+                .startSpan();
+        return withTracing(span, () -> delegate.put(pageId, byteBuffer, cacheContext));
+    }
+
+    @Override
+    public int get(PageId pageId, int position, int length, ReadTargetBuffer readTargetBuffer, CacheContext cacheContext)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.get")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .setAttribute(CACHE_FILE_READ_POSITION, pageId.getPageIndex() * length)
+                .setAttribute(CACHE_FILE_READ_SIZE, (long) length)
+                .startSpan();
+        return withTracing(span, () -> delegate.get(pageId, position, length, readTargetBuffer, cacheContext));
+    }
+
+    @Override
+    public int getAndLoad(PageId pageId, int position, int length, ReadTargetBuffer readTargetBuffer, CacheContext cacheContext, Supplier<byte[]> supplier)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.getAndLoad")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .setAttribute(CACHE_FILE_READ_POSITION, (long) position)
+                .setAttribute(CACHE_FILE_READ_SIZE, (long) length)
+                .startSpan();
+        return withTracing(span, () -> delegate.getAndLoad(pageId, position, length, readTargetBuffer, cacheContext, supplier));
+    }
+
+    @Override
+    public void deleteFile(String fileId)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.deleteFile")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .startSpan();
+        withTracing(span, () -> delegate.deleteFile(fileId));
+    }
+
+    @Override
+    public void deleteTempFile(String fileId)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.deleteTempFile")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .startSpan();
+        withTracing(span, () -> delegate.deleteTempFile(fileId));
+    }
+
+    @Override
+    public boolean delete(PageId pageId)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.delete")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .startSpan();
+        return withTracing(span, () -> delegate.delete(pageId));
+    }
+
+    @Override
+    public boolean append(PageId pageId, int position, byte[] bytes, CacheContext cacheContext)
```

Comment:
Interesting, I did not know it was possible to append to a page. This can potentially be used to reduce the size of the external reads we make.